src/output/outputMod.F90

```diff
@@ -2166,27 +2168,118 @@ subroutine storeSurfsolInBuffer(sps, buffer, nn, blockID, &
         end do
 
         case (cgnsSepSensor)
+        if (sepmodel == surfvec) then
+            do j = rangeFace(2, 1), rangeFace(2, 2)
+                if (present(jBeg) .and. present(jEnd) .and. (useRindLayer)) then
+                    jor = j + jBegOr - 1
+                    if (jor == jBeg) then
+                        jj = j + 1
+                    else if (jor == jEnd + 1) then
+                        jj = j - 1
+                    else
+                        jj = j
+                    end if
+                else
+                    jj = j
 
-        do j = rangeFace(2, 1), rangeFace(2, 2)
-            do i = rangeFace(1, 1), rangeFace(1, 2)
-                nn = nn + 1
+                end if
+
+                do i = rangeFace(1, 1), rangeFace(1, 2)
+
+                    if (present(iBeg) .and. present(iEnd) .and. (useRindLayer)) then
+                        ior = i + iBegor - 1
+                        if (ior == iBeg) then
+                            ii = i + 1
+                        else if (ior == iEnd + 1) then
+                            ii = i - 1
+                        else
+                            ii = i
+                        end if
+                    else
+                        ii = i
+                    end if
+
+                    nn = nn + 1
 
-                ! Get normalized surface velocity:
-                v(1) = ww2(i, j, ivx)
-                v(2) = ww2(i, j, ivy)
-                v(3) = ww2(i, j, ivz)
+                    ! Get normalized surface velocity:
+                    v(1) = ww2(ii, jj, ivx)
+                    v(2) = ww2(ii, jj, ivy)
+                    v(3) = ww2(ii, jj, ivz)
 
-                ! Normalize
-                v = v / (sqrt(v(1)**2 + v(2)**2 + v(3)**2) + 1e-16)
+                    ! Normalize
+                    v = v / (sqrt(v(1)**2 + v(2)**2 + v(3)**2) + 1e-16)
+                    mm = viscPointer(ii, jj)
 
-                ! Dot product with free stream
-                sensor = -dot_product(v, velDirFreeStream)
+                    norm(1) = BCData(mm)%norm(ii, jj, 1)
+                    norm(2) = BCData(mm)%norm(ii, jj, 2)
+                    norm(3) = BCData(mm)%norm(ii, jj, 3)
 
-                !Now run through a smooth heaviside function:
-                sensor = one / (one + exp(-2 * sepSensorSharpness * (sensor - sepSensorOffset)))
-                buffer(nn) = sensor
+                    vectNormProd = velDirFreeStream(1) * norm(1) + &
+                                   velDirFreeStream(2) * norm(2) + &
+                                   velDirFreeStream(3) * norm(3)
+
+                    vectNorm(1) = velDirFreeStream(1) - vectNormProd * norm(1)
+                    vectNorm(2) = velDirFreeStream(2) - vectNormProd * norm(2)
+                    vectNorm(3) = velDirFreeStream(3) - vectNormProd * norm(3)
+
+                    vectNorm = vectNorm / (sqrt(vectNorm(1)**2 + vectNorm(2)**2 + &
```

Comment:
The variable names here are now inconsistent with the variables in surfaceIntegrations.F90